collector/bonding_linux.go

```diff
@@ -67,6 +67,9 @@ func readBondingStats(root string) (status map[string][2]int, err error) {
 	status = map[string][2]int{}
 	masters, err := ioutil.ReadFile(path.Join(root, "bonding_masters"))
 	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
`procPath` is global variable. Kinda ugly, but you could change that (and revert that change in a defer call) and call the `Update` method itself. I don't have a better idea right now.